### PR TITLE
Add ability to initiate connection from server

### PR DESCRIPTION
**Pull request by SerhiiGadzhilov**
SYNERGY1-1609

### CODE REVIEW

File: src/gui/src/MainWindow.cpp [hunk 1]
```diff
@@ -947,6 +955,14 @@ bool MainWindow::serverArgs(QStringList& args, QString& app)
         return false;
     }
 
+    if (appConfig().getServerClientMode() &&
+        m_pLineEditClienIp->text().isEmpty())
+    {
+        QMessageBox::warning(this, tr("Client IP address or name is empty"),
+                             tr("Please fill in a client ip address or name."));
```
Comment (nbolton) [inline src/gui/src/MainWindow.cpp:962]: Change to "IP address"

File: src/gui/src/MainWindowBase.ui [hunk 2]
```diff
@@ -7,7 +7,7 @@
     <x>0</x>
     <y>0</y>
     <width>720</width>
-    <height>658</height>
+    <height>552</height>
```
Comment (nbolton) [inline src/gui/src/MainWindowBase.ui:10]: @SerhiiGadzhilov Was this intentional? Could you attach a screenshot of how it looks now?

File: src/gui/src/SettingsDialogBase.ui [hunk 3]
```diff
@@ -7,7 +7,7 @@
     <x>0</x>
     <y>0</y>
     <width>404</width>
-    <height>620</height>
+    <height>819</height>
```
Comment (nbolton) [inline src/gui/src/SettingsDialogBase.ui:10]: @SerhiiGadzhilov Quite a big change, screenshot please.

File: src/lib/net/InverseSockets/AutoArchSocket.cpp [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+    if (isValid()) {
+        ARCH->setReuseAddrOnSocket(m_socket, value);
+    }
+}
+
+void AutoArchSocket::closeSocket()
+{
+    if (isValid()) {
+        try {
+            LOG((CLOG_DEBUG "Closing socket: %08X", m_socket));
```
Comment (nbolton) [inline src/lib/net/InverseSockets/AutoArchSocket.cpp:59]: Maybe lower case for consistency?

File: src/lib/net/InverseSockets/AutoArchSocket.cpp [hunk 5]
```diff
@@ -0,0 +1,201 @@
...
+#include "base/Log.h"
+#include "arch/Arch.h"
+#include "arch/XArch.h"
+#include "net/XSocket.h"
+
+AutoArchSocket::AutoArchSocket(IArchNetwork::EAddressFamily family)
+{
+    try {
+        m_socket = ARCH->newSocket(family, IArchNetwork::kSTREAM);
+        LOG((CLOG_DEBUG "Opening new socket: %08X", m_socket));
```
Comment (nbolton) [inline src/lib/net/InverseSockets/AutoArchSocket.cpp:29]: Lowe case for consistency

**Review by nbolton — CHANGES_REQUESTED** [review]

File: src/lib/net/InverseSockets/AutoArchSocket.cpp (hunk 4 above)
Comment (SerhiiGadzhilov) [reply src/lib/net/InverseSockets/AutoArchSocket.cpp]: Fixed

File: src/lib/net/InverseSockets/AutoArchSocket.cpp (hunk 5 above)
Comment (SerhiiGadzhilov) [reply src/lib/net/InverseSockets/AutoArchSocket.cpp]: Fixed.
Yes, this is the same trace as in the original TCP class.

File: src/gui/src/MainWindowBase.ui (hunk 2 above)
Comment (SerhiiGadzhilov) [reply src/gui/src/MainWindowBase.ui:10]: I have sent screenshot

File: src/gui/src/SettingsDialogBase.ui (hunk 3 above)
Comment (SerhiiGadzhilov) [reply src/gui/src/SettingsDialogBase.ui:10]: I have sent sceenshot

File: src/gui/src/MainWindow.cpp (hunk 1 above)
Comment (SerhiiGadzhilov) [reply src/gui/src/MainWindow.cpp]: Fixed

**Review by aquacash5 — APPROVED** [review]

**Pull request merged**